pkg/asset/agent/image/oc.go

```diff
@@ -169,6 +176,62 @@ func (r *release) getImageFromRelease(imageName string) (string, error) {
 
 	return image, nil
 }
+func (r *release) validateReleaseImageArch(architecture string) error {
+	// This requires the 'oc' command so make sure its available
+	_, err := exec.LookPath("oc")
+	var cmd string
+	if err != nil {
+		logrus.Error("Unable to validate release image architecture: \"oc\" command is not found")
+		return err
+	}
+
+	cmd = fmt.Sprintf(templateGetReleaseInfo, r.releaseImage)
+	image, err := execute(r.executer, r.pullSecret, cmd)
+	if err == nil {
+		type ReleaseConfig struct {
+			Architecture string `json:"architecture,omitempty"`
+		}
+		type ReleaseMetaDataMetaData struct {
+			ReleaseArch string `json:"release.openshift.io/architecture,omitempty"`
+		}
+		type ReleaseMetaData struct {
+			Metadata *ReleaseMetaDataMetaData `json:"metadata,omitempty"`
+		}
+		type ReleaseInfo struct {
+			Image    string           `json:"image"`
+			Metadata *ReleaseMetaData `json:"metadata,omitempty"`
+			Config   *ReleaseConfig   `json:"config,omitempty"`
+		}
+		var releaseimagedata ReleaseInfo
+		err := json.Unmarshal([]byte(image), &releaseimagedata)
+		if err != nil {
+			logrus.Errorf("can't read oc adm release info for %s ", r.releaseImage)
+			return err
+		}
+
+		// Check the release image to see if it is multi.
+		if releaseimagedata.Metadata.Metadata != nil {
+			if releaseimagedata.Metadata.Metadata.ReleaseArch == "multi" {
+				err = fmt.Errorf("multi architecture release image %s is not supported", r.releaseImage)
+				return err
+			}
+		} else {
+			// if the release image isn't multi, then its single arch, and it must match the cpu architecture.
+			switch releaseimagedata.Config.Architecture {
```

Comment:
nit: this is really an `if` statement disguised as a `switch` :slightly_smiling_face: